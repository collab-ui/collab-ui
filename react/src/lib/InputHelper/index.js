/** @component input */

import React from 'react';
import PropTypes from 'prop-types';

const InputHelper = ({ message, className }) => {
  return <p className={`cui-input__help-text ${className}`}>{message}</p>;
};

InputHelper.propTypes = {
  /** @prop Optional css class name | '' */
  className: PropTypes.string,
  /** @prop Input help message for parent Input | null */
  message: PropTypes.string.isRequired
};

InputHelper.defaultProps = {
  className: '',
  message: null,
};

InputHelper.displayName = 'InputHelper';

export default InputHelper;

/**
* @component input
* @section Input Helper
*
* @react

import {
  Checkbox,
  InputHelper
} from '@collab-ui/react';

export default function InputHelperDefault() {
  return (
    <div className='row'>
      <Checkbox
        value='us'
        label='us'
        htmlId='inputHelper1'
        onChange={()=>{}}
      >
        <InputHelper message={'I\'m here to help'} />
      </Checkbox>
    </div>
  );
}

**/
