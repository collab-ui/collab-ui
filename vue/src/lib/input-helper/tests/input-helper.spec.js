import { shallowMount } from '@vue/test-utils';
import InputHelper from '../index.vue';

describe('InputHelper', () => {
  it('should match snapshot', () => {
    const wrapper = shallowMount(InputHelper, {
      propsData: {
        message: 'test'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render input help with correct class', () => {
    const container = shallowMount(InputHelper, {
      propsData: {
        message: 'test'
      }
    });

    expect(container.find('p').classes('md-input__help-text')).toEqual(true);
  });

  it('should render message', () => {
    const container = shallowMount(InputHelper, {
      propsData: {
        message: 'test'
      }
    });

    expect(container.find('p').text()).toEqual('test');
  });
});

