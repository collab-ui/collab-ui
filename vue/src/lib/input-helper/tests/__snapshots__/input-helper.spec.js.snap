// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputHelper should match snapshot 1`] = `<p class="md-input__help-text">test</p>`;
